controllers/reconcilers/kserve_metrics_servicemonitor_reconciler.go

```diff
@@ -67,9 +77,36 @@ func (r *KserveMetricsServiceMonitorReconciler) Reconcile(ctx context.Context, l
 	return nil
 }
 
-// TODO remove this reconcile loop in future versions
-func (r *KserveMetricsServiceMonitorReconciler) createDesiredResource(isvc *kservev1beta1.InferenceService) (*v1.ServiceMonitor, error) {
-	return nil, nil
+func (r *KserveMetricsServiceMonitorReconciler) createDesiredResource(ctx context.Context, log logr.Logger, isvc *kservev1beta1.InferenceService) (*v1.ServiceMonitor, error) {
+	isvcRuntime, err := utils.FindSupportingRuntimeForISvc(ctx, r.client, log, isvc)
+	if err != nil {
+		return nil, err
+	}
+
+	desiredServiceMonitor := &v1.ServiceMonitor{
```

Comment:
I'm very confused about including metrics in this PR. This wasn't here on my previous review. Also, the Jiras listed in the PR description don't include any Jiras about metrics.

Do we really want metrics in this PR?